nucypher-core-python/src/lib.rs

```diff
@@ -273,10 +299,10 @@ impl EncryptedKeyFrag {
     ) -> Self {
         Self {
             backend: nucypher_core::EncryptedKeyFrag::new(
-                &signer.backend,
-                &recipient_key.backend,
+                signer.as_ref(),
+                recipient_key.as_ref(),
                 &hrac.backend,
-                verified_kfrag.backend.clone(),
+                verified_kfrag.as_ref().clone(),
```

Comment:
Why does it need to be cloned if it's a reference?

Reply:
The result is actually an object (`VerifiedKeyFrag`), not a reference to it. It is a bit of an implicit machinery called [autoderef](https://stackoverflow.com/questions/28519997/what-are-rusts-exact-auto-dereferencing-rules/28552082#28552082). Basically, Rust first tries to do `clone()` on `&VerifiedKeyFrag`, which results in `&VerifiedKeyFrag`, but that does not satisfy the boundary conditions (the method where this value is passed to takes `VerifiedKeyFrag`). So then it dereferences `&VerifiedKeyFrag` to `VerifiedKeyFrag` and tries `clone()` on that - and that works.